compiler/lib/javascript.ml

```diff
@@ -163,10 +163,10 @@ and binop =
   | NotEq
   | EqEqEq
   | NotEqEq
-  | Lt
-  | Le
-  | Gt
-  | Ge
+  | Lt of bool
```

Comment:
What is the bool here for?

Reply:
This is a hack I used to know if the operator is comparing ints or something else.
Knowing that we're are comparing ints (and in particular, not floats) allows us do the following rewrite
`! ( a < b)` => `a >= b` (which would be wrong if `a` or `b` was `NaN`).
Ideally, such optimization should be done at an earlier stage. If we decide to keep the current approach, we should probably use inline records to explain the meaning of the Boolean.